x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/inference/TrainedModelConfigTests.java

```diff
@@ -307,65 +306,12 @@ public void testSerializationWithCompressedLazyDefinition() throws IOException {
             .test();
     }
 
-    public void testIsAvailableWithLicense() {
+    public void testIsAvailableWithLicenseWillDelegate() {
         TrainedModelConfig.Builder builder = createTestInstance(randomAlphaOfLength(10));
         XPackLicenseState licenseState = mock(XPackLicenseState.class);
-
-        // Reject everything
-        when(licenseState.isAllowedByLicense(any(License.OperationMode.class), anyBoolean(), anyBoolean())).thenAnswer(
-            invocationOnMock -> {
-                final Object[] arguments = invocationOnMock.getArguments();
-                assertTrue((boolean) arguments[1]); // ensure the call is made to require active license
-                return false;
-            }
-        );
-        assertFalse(builder.setLicenseLevel(License.OperationMode.ENTERPRISE.description()).build().isAvailableWithLicense(licenseState));
-        assertFalse(builder.setLicenseLevel(License.OperationMode.PLATINUM.description()).build().isAvailableWithLicense(licenseState));
-        assertFalse(builder.setLicenseLevel(License.OperationMode.GOLD.description()).build().isAvailableWithLicense(licenseState));
-        // Basic license always works not matter what
-        assertTrue(builder.setLicenseLevel(License.OperationMode.BASIC.description()).build().isAvailableWithLicense(licenseState));
-    }
-
-    public void testActivePlatinumLicenseAlwaysWorks() {
-        TrainedModelConfig.Builder builder = createTestInstance(randomAlphaOfLength(10));
-        XPackLicenseState licenseState = mock(XPackLicenseState.class);
-
-        when(licenseState.isAllowedByLicense(License.OperationMode.PLATINUM)).thenReturn(true);
-
-        // Active Platinum license functions the same as Enterprise license (highest) and should always work
-        when(licenseState.isAllowedByLicense(any(License.OperationMode.class), anyBoolean(), anyBoolean())).thenAnswer(
-            invocationOnMock -> {
-                final Object[] arguments = invocationOnMock.getArguments();
-                assertEquals(License.OperationMode.PLATINUM, arguments[0]);
-                assertTrue((boolean) arguments[1]); // ensure the call is made to require active license
-                assertTrue((boolean) arguments[2]);
-                return true;
-            }
-        );
-        assertTrue(builder.setLicenseLevel(License.OperationMode.ENTERPRISE.description()).build().isAvailableWithLicense(licenseState));
```

Comment:
The licensing for inference models is not straightforward as different models can be licensed at different levels can we keep these assertions please so we know exactly what the behaviour is. The test that this code calls another piece of code is not sufficient. And for the gold tests below please

Reply:
These assertions were removed because the method to be tested was reduced to just a single line which simply makes a call to `XPackLicenseState#isAllowedByLicense(licenseLevel)`.  Since tests use mock object, there really was no other logic to be tested for this method. 

But if we add some of the original logic back, e.g. the platinum license check as discussed above, the relevant tests are then become necessary again and need to be added back as well.

Reply:
I know it's a one liner and the mocking makes the test awkward but it may not always be a one-liner if someone refactors the function and I'd prefer to have the safety net of asserting  the expected behaviour. 

That said, the best kind of code is deleted code so why not remove the method completely and call directly and `XPackLicenseState#isAllowedByLicense()` directly then the tests can also go.

Reply:
This is a great suggestion. I updated the code accordingly.